packages/data-context/src/sources/migration/codegen.ts

```diff
@@ -202,18 +206,34 @@ function formatObjectForConfig (obj: Record<string, unknown>) {
   return JSON.stringify(obj, null, 2).replace(/^[{]|[}]$/g, '') // remove opening and closing {}
 }
 
-function createE2eTemplate (pluginPath: string, hasPluginsFile: boolean, options: Record<string, unknown>) {
-  const requirePlugins = `return require('./${pluginPath}')(on, config)`
+function createE2ETemplate (pluginPath: string, createConfigOptions: CreateConfigOptions, options: Record<string, unknown>) {
+  if (!createConfigOptions.hasPluginsFile) {
+    return dedent`
+      e2e: {
+        setupNodeEvents(on, config) {}
+      }
+    `
+  }
+
+  const pluginFile = fs.readFileSync(path.join(createConfigOptions.projectRoot, pluginPath), 'utf8')
+  const hasExportDefault = pluginFile.includes('export default')
```

Comment:
This would be more foolproof I believe

```suggestion
  const hasExportDefault = /\nexport default /.test(pluginsFile)
```

We could also create a harmonizing function inside of cypress to use instead of require.

```
import { compatibleRequire } from 'cypress'
```

TBH, even the use of the `require` function at all seems like a bad idea.
An `import` at the top of the file would probably work better.

Reply:
One of the nice things with `require` is if there are any side effects, it won't execute until `setupNodeEvents` is called - that's closer to the original behavior.

Also couldn't you have

```js
      export default (on, config) {}
```

And your regexp would not work?

Reply:
Also couldn't you have

```js
      export default (on, config) {}
```

And your regexp would not work?

Reply:
For the regexp, I am afraid of people using "task" to scaffold files before they start testing.

This could really throw you off. 

My all-time fav is the babel way:

```js
export default defineConfig({
  e2e: setupNodeEvent(on, config){
    const pluginsFunction = require('./cypress/plugins')
    return pluginsFunction.default ? pluginsFunction.default(on, config) : pluginsFunction(on, config)
  }
})

Reply:
I don't think we want to add a conditional to the generated config; it seems weird that we can't just give them the correct code out of the box.

Reply:
I also don't want to actually import their file to see if it has a `.default` property since that would trigger any side effects during config generation, not really ideal.

Reply:
Ok, I ended up implementing this with babel.